Adding a use cases document

arschles opened the pull request:
This PR adds a document describing use cases for the HTTP add on.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Fixes #81

arschles commented:
@tomkerkhove we discussed this document in the March 2, 2021 developer standup. I've submitted this PR as a draft. Can you please take a look and let me know if this is the kind of thing you had in mind?

tomkerkhove commented on docs/use_cases.md:
```diff
@@ -0,0 +1,31 @@
+
+In this use case, an application may be containerized running on a managed cloud platform that supports containers. Below is a non-exhaustive, non-ordered list of some examples:
+
+- [Azure App Services](https://docs.microsoft.com/en-us/azure/app-service/quickstart-custom-container?pivots=container-linux)
```
Nit - Let's alphabetically sort to not be biased

tomkerkhove reviewed (commented):
I think it's a good start but would add some schematics of the Kubernetes resources that comes into play and use color-coding who's in charge of what.

Would that make sense?

arschles commented:
@tomkerkhove I think so. I'll write that for one of these use cases and ask you for another review

arschles commented:
@tomkerkhove I added some color coding. I'm not exactly sure what you mean by "schematics" of Kubernetes resources though. Do you mean a YAML reference (if so, I added that in https://github.com/kedacore/http-add-on/pull/78/files)

tomkerkhove commented on docs/use_cases.md:
```diff
@@ -0,0 +1,39 @@
+
+- [Amazon ECS](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/Welcome.html)
+- [Azure App Services](https://docs.microsoft.com/en-us/azure/app-service/quickstart-custom-container?pivots=container-linux)
+- - [Digital Ocean App Platform](https://www.digitalocean.com/products/app-platform/)
```
```suggestion
- [Digital Ocean App Platform](https://www.digitalocean.com/products/app-platform/)
```

tomkerkhove reviewed (approved):
LGTM, the schematics I was referring to was the one you've added with color-coding.

Eventually we should expand the use cases with service mesh, and all once we support it

arschles commented:
@tomkerkhove wonderful. And, agreed. I added a comment at https://github.com/kedacore/http-add-on/issues/6#issuecomment-801237691 reminding us that we need to document how service meshes integrate with the HTTP add on, when we do implement it.
